query-engine/connectors/sql-query-connector/src/cursor_condition.rs

```diff
@@ -78,32 +125,45 @@ pub fn build(query_arguments: &QueryArguments, model: &ModelRef) -> (Option<Tabl
             let cursor_condition = cursor_row.clone().equals(cursor_values.clone());
 
             // Orderings for this query. Influences which fields we need to fetch for comparing order fields.
-            let mut order_definitions = order_definitions(query_arguments, model);
+            let mut definitions = order_definitions(query_arguments, model, &ordering_joins);
 
             // Subquery to find the value of the order field(s) that we need for comparison. Builds part #1 of the query example in the docs.
-            let order_subquery = order_definitions
+            let order_subquery = definitions
                 .iter()
-                .fold(Select::from_table(model.as_table()), |select, (field, _)| {
-                    select.column(field.as_column())
+                .fold(Select::from_table(model.as_table()), |select, definition| {
+                    select.column(
+                        definition
+                            .order_column
+                            .clone()
+                            .alias(definition.field_aliased.1.to_owned()),
+                    )
                 })
                 .so_that(cursor_condition);
 
+            let order_subquery = ordering_joins
+                .iter()
+                .flat_map(|j| &j.joins)
+                .fold(order_subquery, |acc, join| acc.left_join(join.data.clone()));
```

Comment:
We're now cloning the join data here, that's passed as a reference from the `ordering::build` method